### ISSUE TITLE
[UX] Current menu local task (child/button) is unclear

### ISSUE THREAD

**Issue by jenlampton** [backdrop/backdrop-issues#2086]
Which button is active here? 
![screen shot 2016-08-21 at 10 46 39 pm](https://cloud.githubusercontent.com/assets/397895/17844723/424427d4-67f1-11e6-8eca-7837d7bb53c1.png)

For the tabs, it's pretty clear. The white one is active.
Unfortunately, for the sub-tabs, it's the opposite! The gray button is the one that's active.

I would like it to be visually clear which sub-tab is active too. Would switching the gray/white solve the problem? I have no idea. Do we have any designer-folks who could take a stab at improving this situation? 

---
PR from @olafgrabienski https://github.com/backdrop/backdrop/pull/1525

**Comment by klonos** [comment]
I have been puzzled by this a few times as well. I find it confusing that the active/inactive colors swap for primary and secondary tabs, but then again the primary tabs are on a gay background while the secondary tabs on a white one. Hmm.

**Comment by klonos** [comment]
...it's less confusing when there are more than two secondary tabs of course.

**Comment by klonos** [comment]
How about this? Changed the inactive secondary tabs to follow similar pattern to the inactive primary tabs:

1. font weight set to normal (only *active* secondary tab is bold)
2. font color set to the lighter `#757575` (fades to `#000` on hover)

**Comment by jenlampton** [comment]
I think maybe having the active colors match might help too. If active is lighter in the primary tabs, can we try lighter (and yes, bold text) for the active secondary?

**Comment by jenlampton** [comment]
Or, maybe having the active tab *not* look like a button would help too. It's a link to the page we are already on, we don't want people to click it. What if it were square or something?  I'm so bad at designy stuff :(

**Comment by klonos** [comment]
>  I'm so bad at designy stuff :(

Me too. @wesruv ?? Some input please?

**Comment by olafgrabienski** [comment]
> Which button is active here?

These elements are not intended as buttons but as (secondary) tabs, and I would propose to design them more as tabs to solve the issue. I like the design of the [Shiny](https://www.drupal.org/project/shiny) theme in Drupal 7, so I had a look how they designed the secondary tabs and adapted their design to Backdrop. See the screenshots below.

**Drupal, Shiny theme:**

![screen-drupal-content-type-manage-display-shiny](https://cloud.githubusercontent.com/assets/14188307/17888999/3f4a5ca4-692e-11e6-8192-daa405cff962.png)


**Backdrop, like Shiny:**

![screen-backdrop-content-type-manage-display-seven-like-drupal-shiny](https://cloud.githubusercontent.com/assets/14188307/17888977/17f55730-692e-11e6-82c9-c8e5b83fd81e.jpg)

See the following CSS changes:

```
ul.secondary li {
  padding: 0;
}
ul.secondary li a {
  border: 0;
  border-radius: 3px;
  background: #ddd;
  color: #000;
  font-weight: normal;
}
ul.secondary li.active a,
ul.secondary li.active a.active {
  color: #fff;
  background: #666;
}

```
What do you think about the adaption?

PS: There are still some differences, e.g. the `float: right` in Shiny (which we shouldn't copy) and the bigger font size of the secondary tabs in Shiny (looks better in my opinion).

**Comment by jenlampton** [comment]
Yes! I like this better :)

- -
Please excuse brevity and typos. Message typed with my thumbs from an Open
Source OS.

**Comment by olafgrabienski** [comment]
Thanks for your feedback @jenlampton! Other opinions?

Regarding the bigger font size, I propose to add it to the CSS as well:

```
ul.secondary {
  font-size: 0.875em;
}
```

**Comment by quicksketch** [comment]
This is much better @olafgrabienski! I like that the font and background for "Inactive" matches between primary and secondary tabs. For the active tabs, I also like the matching font weight. That seems like a no-brainer as well. Both these changes could be easily justified from a UX consistency standpoint.

I am not sure about the font and background color for the active tab, but it seems better to me.

The funny thing about about squaring the tabs is that at least in the Backdrop Seven theme, that makes them look *more* like our buttons:

![selection_165](https://cloud.githubusercontent.com/assets/100206/17994537/6a17cbe2-6b0e-11e6-9a95-471d383760fd.png)

@klonos made a PR at https://github.com/backdrop/backdrop/pull/1496 with the first approach, but I don't think the fix is as effective as Olaf's suggestion.

**Comment by Bojhan** [comment]
There are a few other ways to make them not look like buttons, by attaching them visually:

1) Have no margin between them and only separate the tabs using borders.
2) Have a border-bottom connecting the two (like Zen).
3) Position them vertically (e.g. using a sidebar).

Additionally you could also have a background color on the entire bar (I know in Seven this was hard to do). 

Overall its worthwhile to explore using a "active" indicator beyond just bold or background. Using for example a small bar:

<img width="165" alt="screen shot 2016-08-27 at 21 49 18" src="https://cloud.githubusercontent.com/assets/209807/18029720/323bacb4-6ca0-11e6-8294-f9a7ad9f298c.png">

**Comment by jenlampton** [comment]
Ooh, I like that blue! That would be inline with our active state on the
vertical tabs too.

- -
Please excuse brevity and typos. Message typed with my thumbs from an Open
Source OS.

**Comment by klonos** [comment]
> The funny thing about squaring the tabs is that at least in the Backdrop Seven theme, that makes them look more like our buttons:

My thought exactly.

> @klonos made a PR at backdrop/backdrop#1496 with the first approach, but I don't think the fix is as effective as Olaf's suggestion.

I'll draft another PR or two with suggestions made here so we can compare and see what works best.

> Ooh, I like that blue! That would be inline with our active state on the vertical tabs too.

Same here.

**Comment by wesruv** [comment]
I find the issue being multiple layers of tabs, it's bad UI. They're a reference to manilla folders IRL, and I have yet to see nested manilla folders :stuck_out_tongue:

**Comment by wesruv** [comment]
Oops, pressed Ctrl + Enter when I made my smiley instead of just enter... d'oh!

That said, I don't know what to do with the damn things.
I have a module that fixes them to the bottom of the screen (which I'm thinking isn't a great idea nowadays) and I handled the second row like so:
![image](https://cloud.githubusercontent.com/assets/5607236/18038141/d1e11fe4-6d61-11e6-95ef-23f01bcde189.png)

I think the idea is to have a second layer of tabs that's still 'tabby', but looks distinct from the first row.

**Comment by wesruv** [comment]
I'm not sure how I feel about this mockup, but figured it's worth sharing. This is a screenshot of Basis, but the secondary tabs are made up.

![image](https://cloud.githubusercontent.com/assets/5607236/18038336/d7e0eeee-6d64-11e6-81cf-27719f64a119.png)

Might try out some other ideas later...

**Comment by quicksketch** [comment]
My preference on this is to "fix the bug" in Seven by implementing something as suggested by @olafgrabienski in https://github.com/backdrop/backdrop-issues/issues/2086#issuecomment-241692868. Even just the font/background color swap is a great start. Reworking the tabs entirely from a design standpoint should probably be a separate task.

**Comment by olafgrabienski** [comment]
Same opinion as @quicksketch: Let's fix the look in a reasonable way for the moment.

If it's better to style the second tab layers even more 'tabbish' like @wesruv suggested, or if we should question multiple tab layers at all, could be a separate issue.

**Comment by olafgrabienski** [comment]
Seems not decided yet: Will the fix in https://github.com/backdrop/backdrop-issues/issues/2086#issuecomment-241692868 be part of 1.5?

**Comment by klonos** [comment]
How about something like this then? Blue border-top for active primary tab to match our vertical tabs, secondary tabs reduced to text + border-bottom (definitely makes them look less button-y) with active one getting black border color:

![backdrop-issue2086-mockup](https://cloud.githubusercontent.com/assets/2423362/18318439/0a5a5042-7565-11e6-95b9-9f44590630f4.gif)

**Comment by klonos** [comment]
...no PR yet, the screenshot above is a hacked mockup. This is what I actually have so far:

![backdrop-issue2086-actual](https://cloud.githubusercontent.com/assets/2423362/18318608/a7aba3d2-7565-11e6-920e-a9cb23c7110b.gif)

Need to get some 💤 ...let me know what you think people and I'll see if I can finish this. Unless someone else beats me to it (@wesruv) 😄

**Comment by jenlampton** [comment]
I think we should focus on the small change suggested in https://github.com/backdrop/backdrop-issues/issues/2086#issuecomment-241692868 for 1.5 and do the larger change with the blues (@klonos I love where this is headed!) in 1.6, or just any time after 1.5.  We need more time for discussion / iteration if we are going to change the design that much, and we don't really have it right now, and need to be focused on some of these other issues.

Does anyone want to make us a new PR?

**Comment by olafgrabienski** [comment]
I've only little experience with Git and no experience with the GitHub workflow at all (besides commenting issues). So, I guess I'm not able to create a pull request at the moment.

I'm however curious to learn the GitHub workflow and would like to be able to make PRs for Backdrop.  Found a topic in the GitHub help, https://help.github.com/articles/creating-and-deleting-branches-within-your-repository. Is that the proper place to learn, or do you recommend other sources in that matter? And is there a particular Backdrop way to make PRs? (Finally: How do you create the related test environments like http://1523.backdrop.backdrop.qa.backdropcms.org?)

**Comment by klonos** [comment]
Hey there @olafgrabienski I can help with that...

**Comment by olafgrabienski** [comment]
Great, how would we start?

**Comment by klonos** [comment]
...sec. I'm typing 😄

**Comment by olafgrabienski** [comment]
@klonos, thanks, take your time! Have to take some sleep now ...

**Comment by klonos** [comment]
Hold on @olafgrabienski, the text was getting rather long and would be OT if posted here. Besides, I thought that having it in the backdropcms.org website as a blog post would help others, so:

https://backdropcms.org/news/creating-pull-requests-against-core-easy-noob-way

@jenlampton is proof-reading and making corrections at the moment, so it's unpublished. Once ready, you'll be have a guide. Soon as a get a chance, I'll improve it with screenshot to make things even easier for beginners 😉

**Comment by klonos** [comment]
...just saw your comment now. Sleep tight and by the time you're up, the post will be waiting for you 😉

**Comment by klonos** [comment]
> ...having it in the backdropcms.org website as a blog post would help others too

...plus @jenlampton will get her weekly blog post 😉

**Comment by klonos** [comment]
@olafgrabienski, perhaps you'd like to join us on [Gitter](https://gitter.im/backdrop/backdrop-issues)? That's where we catch up and discuss Backdrop. You can ask pretty much anything that you are concerned with, be it code or just random subjects related to Drupal/Backdrop.

**Comment by quicksketch** [comment]
Great post @klonos!

**Comment by olafgrabienski** [comment]
Thank you very much for the post on backdropcms.org! Didn't try it out yet, but seems very clear and will definitely be helpful! In case of questions I'll join you on Gitter.

Regarding actually making the PR: I'll try it but need some time because of other duties ...

**Comment by olafgrabienski** [comment]
Well, I've made the PR! Managed the GitHub workflow with the kind help of https://backdropcms.org/news/creating-pull-requests-against-core-easy-noob-way written by @klonos, thanks again! (Found a few typos btw, where do I post them?)

All checks have passed, and I believe the first try was already fine. Forgot only to address hovering on the secondary tab links in https://github.com/backdrop/backdrop-issues/issues/2086#issuecomment-241692868 and added hover in a 2nd try. Can someone review it?

**Comment by jenlampton** [comment]
@olafgrabienski the PR looks great!  I added one note about needing to include vendor prefixes for `border-radius` but other than that this looks good to go for me.
![screen shot 2016-09-08 at 4 26 39 pm](https://cloud.githubusercontent.com/assets/397895/18370349/37c54722-75e1-11e6-9349-4a1072b4f5ec.png)

**Comment by olafgrabienski** [comment]
Perfect! Will add the vendor prefixes (tomorrow, UTC+2h).

**Comment by olafgrabienski** [comment]
I have added vendor-prefixes for border-radius, checks have passed.

Reviews welcome: https://github.com/backdrop/backdrop/pull/1525.

**Comment by olafgrabienski** [comment]
@jenlampton - Don't know the further workflow. Do I have to do something more to get the fix in? Works fine in my opinion.

**Comment by jenlampton** [comment]
@olafgrabienski The only blocker here is a final review (by someone other than you, sorry!) Maybe @klonos or @docwilmot will get to it before me? I'll be back on backdrop duty in a few hours, and can check on it then if no one else has :)

**Comment by olafgrabienski** [comment]
> a final review (by someone other than you, sorry!)

Sure!

**Comment by klonos** [comment]
Regarding vendor prefixes, do we also add `-ms-border-radius` and `-o-border-radius`?

Other than that looks good to me, but would probably be best if an expert gave it a quick look. @wesruv? ...I know there's a lot on your plate with Basis and the rest of the things we're trying to get in for 1.5.0, but this change is like only a dozen of lines.

**Comment by klonos** [comment]
ALSO, it seems to me like we we focusing on the secondary tabs, that we missed the fact that the "Configure" button on that page is not rendered as a dropbutton action. What's up with that?

**Comment by klonos** [comment]
> ...the "Configure" button on that page is not rendered as a dropbutton action.

**Comment by olafgrabienski** [comment]
> Regarding vendor prefixes, do we also add -ms-border-radius and -o-border-radius?

Perhaps we should, at least there are some `-ms` and `-o` prefixes in the `style.css` of the Seven theme. Until now `border-radius` is missing the mentioned prefixes in the file, that's why I didn't add them in the PR. In my opinion that should be a separate and maybe broader issue like "Refactor style.css".

**Comment by olafgrabienski** [comment]
> ... we missed the fact that the "Configure" button on that page is not rendered as a dropbutton action. What's up with that?

That's also a good candidate for another issue, isn't it?

**Comment by serundeputy** [comment]
>... we missed the fact that the "Configure" button on that page is not rendered as a dropbutton action. What's up with that?

yes let's make a new issue for it

**Comment by klonos** [comment]
...yes 😉

**Comment by jenlampton** [comment]
This looks good to me!  Thanks @olafgrabienski

**Comment by quicksketch** [comment]
So much better, thanks @olafgrabienski!

![screenshot from 2016-09-15 15-53-49](https://cloud.githubusercontent.com/assets/100206/18570159/a9756f66-7b5c-11e6-80aa-b611ecae81f9.png)

Merged this into 1.x for 1.5.0. I decided not to include it in 1.4.x. If anyone feels strongly about needing to backport it we can, but as this changes the actual appearance I thought it would be better to keep it in the next version instead of affecting (the very few sites) that going to be using 1.4.5 instead of jumping to 1.5.0.
